### ISSUE TITLE
about inference

### ISSUE THREAD

**Issue by GallonDeng** [Kurumi233/Mutual-Channel-Loss#3]
nice work! the code is clean, however, is it possible to integate the basemodel and metric module and provide a script for loading the saved pth model and doing inference? It would be convenient for demo or production.

**Comment by Kurumi233** [comment]
> nice work! the code is clean, however, is it possible to integate the basemodel and metric module and provide a script for loading the saved pth model and doing inference? It would be convenient for demo or production.

I just uploaded a test file, maybe it can help you.

**Comment by GallonDeng** [comment]
Thanks for your update. It's better.
 I tried to define a new module with both BaseModel and metric part, and load the saved pth into the new fused model through rewriting the state_dict. In the end, one can just load the resaved state_dict into the fused model to do inference. It's simple and maybe you can try it.

**Comment by Kurumi233** [comment]
> Thanks for your update. It's better.
> I tried to define a new module with both BaseModel and metric part, and load the saved pth into the new fused model through rewriting the state_dict. In the end, one can just load the resaved state_dict into the fused model to do inference. It's simple and maybe you can try it.

Thanks for your advice.
Actually, separate the two modules can easily control the learning rate, and the training process will be clearly. This work is mainly to integrate the loss function that can be easily used in other works. You can modify the model as needed.
Looks like you've done it.  _(:з」∠)_

**Comment by Kurumi233** [comment]
> Thanks for your update. It's better.
> I tried to define a new module with both BaseModel and metric part, and load the saved pth into the new fused model through rewriting the state_dict. In the end, one can just load the resaved state_dict into the fused model to do inference. It's simple and maybe you can try it.

Here might be a simple way to avoid rewriting, add little code in BaseModel will be ok.
```
class BaseModel(nn.Module):
    def __init__(self, *args):
        """
        other code
        """

        self.metric = None

    def forward(self, x):
        """
        other code
        """

        if self.metric is not None:
            out = self.metric(out)

        """
        return
        """
````

In the inference stage, for example, just let `model.metric = metric` in `test.py` after load the state_dict.
It is a simple way with little changes in this work, but for demo or production, rewriting will be better :)
